config/setup/git.go

```diff
@@ -20,13 +21,26 @@ func Git(c *Controller) (middleware.Middleware, error) {
 		return nil, err
 	}
 
-	c.Startup = append(c.Startup, func() error {
-		// Start service routine in background
-		git.Start(repo)
+	// If a HookUrl is set, we switch to event based pulling.
+	// Install the url handler
+	if repo.HookUrl != "" {
 
-		// Do a pull right away to return error
-		return repo.Pull()
-	})
+		webhook := &webhook.WebHook{Repo: repo}
+		return func(next middleware.Handler) middleware.Handler {
+			webhook.Next = next
+			return webhook
+		}, nil
+
```

Comment:
I would be too. Did you know it only pulls changes? So if the site hasn't changed since the last pull, nothing will happen anyway.